matview/management/commands/refresh_matviews.py

```diff
@@ -1,24 +1,57 @@
+from threading import Lock
+
 from django.core.management import BaseCommand
-from django.utils import timezone
 from django.db.models.signals import post_save
 
 from matview.models import MatView
 
 
+def synchronized(lock):
+    """ Synchronization decorator. """
+
+    def wrap(f):
+        def newFunction(*args, **kw):
```

Comment:
Maybe throw in ``@wraps(f)`` before ``def wrap(f)`` (from functools import wraps) so tracebacks and such look better.